File: app/schemas/location.py
from datetime import date
from pydantic import BaseModel, Field


class PinOut(BaseModel):
    id: int
    lat: float
    lon: float
    address: str
    current_business: str | None = None
    current_category: str | None = None


class TimelineEntry(BaseModel):
    business_name: str
    category: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    is_current: bool


class LocationDetail(BaseModel):
    id: int
    lat: float
    lon: float
    address: str
    timeline: list[TimelineEntry]


class LocationsResponse(BaseModel):
    locations: list[PinOut]
    count: int
    cursor: str | None = None

File: web/src/types/location.ts
export interface Pin {
  id: number
  lat: number
  lon: number
  address: string
  current_business: string | null
  current_category: string | null
}

export interface PinsResponse {
  locations: Pin[]
  count: number
  cursor: string | null
}

export interface TimelineEntry {
  business_name: string
  category: string | null
  start_date: string | null
  end_date: string | null
  is_current: boolean
}

export interface LocationDetail {
  id: number
  lat: number
  lon: number
  address: string
  timeline: TimelineEntry[]
}

export interface LocationsQueryParams {
  west: number
  south: number
  east: number
  north: number
  limit?: number
  cursor?: string
}
